src/GitHubExtension/Widgets/GitHubRepositoryWidget.cs

```diff
@@ -0,0 +1,215 @@
+        Log.Logger()?.ReportDebug(Name, ShortId, $"ActionInvoked: {verb}");
+
+        switch (verb)
+        {
+            case WidgetAction.CheckUrl:
+                HandleCheckUrl(actionInvokedArgs);
+                break;
+
+            default:
+                base.OnActionInvoked(actionInvokedArgs);
+                break;
+        }
+    }
+
+    protected override void ResetWidgetInfoFromState()
+    {
+        var dataObject = JsonNode.Parse(ConfigurationData);
+
+        if (dataObject == null)
+        {
+            return;
+        }
+
+        RepositoryUrl = dataObject["url"]?.GetValue<string>() ?? string.Empty;
+
+        base.ResetWidgetInfoFromState();
+    }
+
+    public override void OnCustomizationRequested(WidgetCustomizationRequestedArgs customizationRequestedArgs)
+    {
+        SavedConfigurationData = ConfigurationData;
+        SetConfigure();
+    }
+
+    private void HandleCheckUrl(WidgetActionInvokedArgs args)
+    {
+        // Set loading page while we fetch data from GitHub.
+        Page = WidgetPageState.Loading;
+        UpdateWidget();
+
+        // This is the action when the user clicks the submit button after entering a URL while in
+        // the Configure state.
+        Page = WidgetPageState.Configure;
+        var data = args.Data;
+        var dataObject = JsonObject.Parse(data);
+        if (dataObject != null && dataObject["url"] != null)
+        {
+            RepositoryUrl = dataObject["url"]?.GetValue<string>() ?? string.Empty;
+
+            ConfigurationData = data;
+
+            var updateRequestOptions = new WidgetUpdateRequestOptions(Id)
+            {
+                Data = GetConfiguration(RepositoryUrl),
+                CustomState = ConfigurationData,
+                Template = GetTemplateForPage(Page),
+            };
+
+            WidgetManager.GetDefault().UpdateWidget(updateRequestOptions);
+        }
+    }
+
+    public string GetConfiguration(string dataUrl)
+    {
+        var configurationData = new JsonObject
+        {
+            { "submitIcon", IconLoader.GetIconAsBase64("arrow.png") },
+        };
+
+        if (dataUrl == string.Empty)
+        {
+            CanPin = false;
+            configurationData.Add("hasConfiguration", false);
+            configurationData.Add("configuring", !CanPin);
+            var repositoryData = new JsonObject
+            {
+                { "url", string.Empty },
+            };
+
+            configurationData.Add("configuration", repositoryData);
+            configurationData.Add("savedRepositoryUrl", SavedConfigurationData);
+            configurationData.Add("saveEnabled", false);
+
+            return configurationData.ToString();
+        }
+        else
+        {
+            try
+            {
+                // Get client for logged in user.
+                var client = GitHubClientProvider.Instance.GetClientForLoggedInDeveloper(true).Result;

```

Comment:
Don't we need to specify the DevId here?

Reply:
While I was doing it, I was thinking that for the two Repository widgets, the developer ID selection is not that necessary, as these widgets get an repository URL as input, so it should track any repository regardless of the dev IDs connected. One change I think makes more sense is to go through all the dev IDs connected until we find one that have access to the repository requested, so this will not be needed to be done by the user.

Reply:
The data code does do this, so I dont think not having one is a blocker. However, I do think it is worth having an offline discussion about the tradeoffs to a selection vs smart-detection of the developer ID.